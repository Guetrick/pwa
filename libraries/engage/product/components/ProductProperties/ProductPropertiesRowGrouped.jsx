import React from 'react';
import PropTypes from 'prop-types';
import { getPropertiesBySubgroup } from './helpers/getPropertiesBySubgroup';
import { ProductPropertiesRow } from './ProductPropertiesRow';

/**
 * @param {Object} props The component props.
 * @returns {JSX}
 */
export function ProductPropertiesRowGrouped({ properties, group }) {
  const filteredProperties = getPropertiesBySubgroup(properties, group);

  return filteredProperties.map(({ label, value }) => (
    <ProductPropertiesRow key={`${label}-${value}`} label={label} value={value} />
  ));
}

ProductPropertiesRowGrouped.propTypes = {
  group: PropTypes.string.isRequired,
  properties: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};
