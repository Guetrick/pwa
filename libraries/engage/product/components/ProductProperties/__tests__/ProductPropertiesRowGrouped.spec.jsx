import React from 'react';
import { shallow } from 'enzyme';
import mockConsole from 'jest-mock-console';
import { ProductPropertiesRowGrouped } from '../ProductPropertiesRowGrouped';

const properties = [
  { subDisplayGroup: 'test1' },
  { subDisplayGroup: 'test1' },
];

describe('<ProductPropertiesRowGrouped />', () => {
  it('not render if no properties match to the group', () => {
    const wrapper = shallow(<ProductPropertiesRowGrouped properties={properties} group="test2" />);
    expect(wrapper.instance()).toEqual(null);
  });

  it('should render two rows of properties', () => {
    const restoreConsole = mockConsole();
    const wrapper = shallow(<ProductPropertiesRowGrouped properties={properties} group="test1" />);
    expect(wrapper.find('ProductPropertiesRow').length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
    restoreConsole();
  });
});
