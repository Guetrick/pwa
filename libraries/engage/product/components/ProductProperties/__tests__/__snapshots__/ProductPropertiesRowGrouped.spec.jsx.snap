// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPropertiesRowGrouped /> should render two rows of properties 1`] = `
Array [
  <ProductPropertiesRow
    key="undefined-undefined"
  />,
  <ProductPropertiesRow
    key="undefined-undefined"
  />,
]
`;
